WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/config/AnalogComplicationConfigRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,754 @@
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.PreviewAndComplicationsConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.UnreadNotificationConfigItem
+import com.example.android.wearable.watchfacekotlin.watchface.AnalogComplicationWatchFaceService
+
+import java.util.concurrent.Executors
+
+/**
+ * Displays different layouts for configuring watch face's complications and appearance settings
+ * (highlight color [second arm], background color, unread notifications, etc.).
+ *
+ *
+ * All appearance settings are saved via [SharedPreferences].
+ *
+ *
+ * Layouts provided by this adapter are split into 5 main view types.
+ *
+ *
+ * A watch face preview including complications. Allows user to tap on the complications to
+ * change the complication data and see a live preview of the watch face.
+ *
+ *
+ * Simple arrow to indicate there are more options below the fold.
+ *
+ *
+ * Color configuration options for both highlight (seconds hand) and background color.
+ *
+ *
+ * Toggle for unread notifications.
+ *
+ *
+ * Background image complication configuration for changing background image of watch face.
+ */
+class AnalogComplicationConfigRecyclerViewAdapter(
+    private val context: Context,
+    watchFaceServiceClass: Class<*>,
+    settingsDataSet: ArrayList<ConfigItemType>
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    /**
+     * Used by associated watch face ([AnalogComplicationWatchFaceService]) to let this
+     * adapter know which complication locations are supported, their ids, and supported
+     * complication data types.
+     */
+    enum class ComplicationLocation {
+        BACKGROUND, LEFT, RIGHT, TOP, BOTTOM
+    }
+
+    // ComponentName associated with watch face service (service that renders watch face). Used
+    // to retrieve complication information.
+    private val watchFaceComponentName: ComponentName
+    private val settingsDataSet: ArrayList<ConfigItemType>
+    var sharedPref: SharedPreferences
+
+    // Selected complication id by user.
+    private var selectedComplicationId: Int
+    private val backgroundComplicationId: Int
+    private val leftComplicationId: Int
+    private val rightComplicationId: Int
+
+    // Required to retrieve complication data from watch face for preview.
+    private val providerInfoRetriever: ProviderInfoRetriever
+
+    // Maintains reference view holder to dynamically update watch face preview. Used instead of
+    // notifyItemChanged(int position) to avoid flicker and re-inflating the view.
+    private lateinit var previewAndComplicationsViewHolder: PreviewAndComplicationsViewHolder
+
+    init {
+        watchFaceComponentName = ComponentName(context, watchFaceServiceClass)
+        this.settingsDataSet = settingsDataSet
+
+        // Default value is invalid (only changed when user taps to change complication).
```

Comment:
If i get it right, you're handling item clicked in the adapter.
This is too much logic for the adapter - the adapter should just display things and delegate to tap handling to another class - the class that holds the RecyclerView.